example/lib/app/widget/app.dart

```diff
@@ -23,47 +23,41 @@ class _AppState extends State<App> {
         routes: <String, WidgetBuilder>{
           AppRoutes.root: (context) => MainPage.withBloc(),
         },
-        theme: ThemeData(
-          useMaterial3: true,
-          scaffoldBackgroundColor: lightThemeData.color.primaryBackground,
-          splashFactory: InkRipple.splashFactory,
-          splashColor: lightThemeData.color.controlHighlight.withOpacity(0.5),
-          highlightColor: Colors.transparent,
-          colorScheme: ColorScheme(
-            brightness: Brightness.light,
-            primary: lightThemeData.color.primary,
-            onPrimary: lightThemeData.color.onPrimary,
-            secondary: lightThemeData.color.secondary,
-            onSecondary: lightThemeData.color.onSecondary,
-            error: lightThemeData.color.error,
-            onError: lightThemeData.color.onError,
-            background: lightThemeData.color.primaryBackground,
-            onBackground: lightThemeData.color.onBackground,
-            surface: lightThemeData.color.surface,
-            onSurface: lightThemeData.color.onSurface,
-          ),
+        theme: _getThemeData(
+          themeData: lightThemeData,
+          brightness: Brightness.light,
         ),
-        darkTheme: ThemeData(
-          useMaterial3: true,
-          scaffoldBackgroundColor: darkThemeData.color.primaryBackground,
-          splashFactory: InkRipple.splashFactory,
-          splashColor: darkThemeData.color.controlHighlight.withOpacity(0.5),
-          highlightColor: Colors.transparent,
-          colorScheme: ColorScheme(
-            brightness: Brightness.dark,
-            primary: darkThemeData.color.primary,
-            onPrimary: darkThemeData.color.onPrimary,
-            secondary: darkThemeData.color.secondary,
-            onSecondary: darkThemeData.color.onSecondary,
-            error: darkThemeData.color.error,
-            onError: darkThemeData.color.onError,
-            background: darkThemeData.color.primaryBackground,
-            onBackground: darkThemeData.color.onBackground,
-            surface: darkThemeData.color.surface,
-            onSurface: darkThemeData.color.onSurface,
-          ),
+        darkTheme: _getThemeData(
+          themeData: darkThemeData,
+          brightness: Brightness.dark,
         ),
       ),
     );
   }
+
+  ThemeData _getThemeData({
+    required AppThemeData themeData,
+    required Brightness brightness,
+  }) {
+    return ThemeData(
```

Comment:
Can't this be ?
```
  ThemeData _getThemeData({
    required Brightness brightness,
  }) {
     var themeData = brightness == Brightness.light ? AppThemeData.light() : AppThemeData.dark()
     ...
  }
```

Reply:
I would like to have explicit `AppThemeData` being passed, because not sure, that we can rely on `Brightness` only. Maybe we could have additional `AppThemeData` in the future (in theory, of course), and we would have `light`, `dark`, `custom` etc.

Reply:
I think that `AppThemeData` most likely will be added in pairs like `customLight` and `customDark` so you can add additional parameter like `useAlternativeTheme` and still resolve `AppThemeData` from within `_getThemeData` method. If there will be more than 2 themes (lets say 5) - then instead of bool `useAlternativeTheme` you can provide theme id or any other identifier as parameter.

Reply:
Reworked.